Tasks/Npm/npmtask.ts

```diff
@@ -1,27 +1,194 @@
+import Q = require('q');
+import fs = require('fs');
 import path = require('path');
+import url = require('url');
 import tl = require('vsts-task-lib/task');
+import trm = require('vsts-task-lib/toolrunner');
+
+interface EnvironmentDictionary { [key: string]: string; }
 
 tl.setResourcePath(path.join( __dirname, 'task.json'));
 
-var npm = tl.createToolRunner(tl.which('npm', true));
+executeTask();
+
+async function executeTask() {
+
+    var cwd = tl.getPathInput('cwd', true, false);
```

Comment:
Double-check this does what you expect in RM, in terms of what you get if the user leaves the field blank